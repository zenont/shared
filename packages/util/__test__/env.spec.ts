import {
  getProcessEnv,
  isDebug,
  isEnv,
  isDevelopment,
  isTest,
  isStaging,
  isProduction
} from '../src'

describe('env', () => {
  const ORIG_ENV = process.env

  beforeEach(() => {
    jest.resetModules()
    delete process.env
  })

  afterEach(() => {
    process.env = ORIG_ENV
  })

  it('should get process env', () => {
    process.env = {
      NODE_ENV: 'production'
    }
    const result = getProcessEnv()
    expect(result).toMatchSnapshot()
  })

  it('should not get process env', () => {
    const result = getProcessEnv()
    expect(result).toMatchSnapshot()
  })

  it('should get process debug', () => {
    process.env = {
      DEBUG: 'true'
    }
    const result = isDebug()
    expect(result).toMatchSnapshot()
  })

  it('should not get process debug', () => {
    process.env = {
      DEBUG: 'false'
    }
    const result = isDebug()
    expect(result).toMatchSnapshot()
  })

  it('should get NODE_ENV development', () => {
    const NODE_ENV = 'development'
    process.env = {
      NODE_ENV
    }

    expect(isEnv(NODE_ENV))
    expect(isDevelopment()).toMatchSnapshot()
    expect(isTest()).toMatchSnapshot()
    expect(isStaging()).toMatchSnapshot()
    expect(isProduction()).toMatchSnapshot()
  })

  it('should get NODE_ENV test', () => {
    const NODE_ENV = 'test'
    process.env = {
      NODE_ENV
    }

    expect(isEnv(NODE_ENV))
    expect(isDevelopment()).toMatchSnapshot()
    expect(isTest()).toMatchSnapshot()
    expect(isStaging()).toMatchSnapshot()
    expect(isProduction()).toMatchSnapshot()
  })

  it('should get NODE_ENV staging', () => {
    const NODE_ENV = 'staging'
    process.env = {
      NODE_ENV
    }

    expect(isEnv(NODE_ENV))
    expect(isDevelopment()).toMatchSnapshot()
    expect(isTest()).toMatchSnapshot()
    expect(isStaging()).toMatchSnapshot()
    expect(isProduction()).toMatchSnapshot()
  })

  it('should get NODE_ENV production', () => {
    const NODE_ENV = 'production'
    process.env = {
      NODE_ENV
    }

    expect(isEnv(NODE_ENV))
    expect(isDevelopment()).toMatchSnapshot()
    expect(isTest()).toMatchSnapshot()
    expect(isStaging()).toMatchSnapshot()
    expect(isProduction()).toMatchSnapshot()
  })
})
