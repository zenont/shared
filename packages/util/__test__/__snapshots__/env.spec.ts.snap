// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`env should get NODE_ENV development 1`] = `true`;

exports[`env should get NODE_ENV development 2`] = `false`;

exports[`env should get NODE_ENV development 3`] = `false`;

exports[`env should get NODE_ENV development 4`] = `false`;

exports[`env should get NODE_ENV production 1`] = `false`;

exports[`env should get NODE_ENV production 2`] = `false`;

exports[`env should get NODE_ENV production 3`] = `false`;

exports[`env should get NODE_ENV production 4`] = `true`;

exports[`env should get NODE_ENV staging 1`] = `false`;

exports[`env should get NODE_ENV staging 2`] = `false`;

exports[`env should get NODE_ENV staging 3`] = `true`;

exports[`env should get NODE_ENV staging 4`] = `false`;

exports[`env should get NODE_ENV test 1`] = `false`;

exports[`env should get NODE_ENV test 2`] = `true`;

exports[`env should get NODE_ENV test 3`] = `false`;

exports[`env should get NODE_ENV test 4`] = `false`;

exports[`env should get process debug 1`] = `true`;

exports[`env should get process env 1`] = `
Object {
  "NODE_ENV": "production",
}
`;

exports[`env should not get process debug 1`] = `false`;

exports[`env should not get process env 1`] = `undefined`;
